doc/HttpTransportAdapter.md

```diff
@@ -0,0 +1,223 @@
+
+## Default HTTP Transport Adapter
+
+The Azure SDK for C++ provides two HTTP transport adapters as part of the azure-core package:
+- The libcurl Transport Adapter
+
+<center>
+
+| Supported Platforms  |
+| --- |
+| Linux |
+| Windows |
+| MacOS |
+| [Others](https://daniel.haxx.se/blog/wp-content/uploads/2020/02/slide-operating-systems.jpg) |
+
+</center>
+
+- The WinHTTP Transport Adapter
+
+<center>
+
+| Supported Platforms  |
+| --- |
+| Windows |
+
+</center>
+
+Azure SDK for C++ users can create and use their own transport adapter [by following the guidance below](#building-a-custom-http-transport-adapter).
+
+For the simplest case, when no specific configuration is set, the default transport adapter will be selected based on the OS. The WinHTTP transport adapter will be used for Windows-based systems (making libcurl an optional dependency). The libcurl transport adapter will be selected for any non-Windows-based system.
+
+## Building HTTP Transport Adapter
+
+The Azure SDK for C++ uses CMake options to define what HTTP transport adapters to build.
+
+You can see [CMake options](https://github.com/Azure/azure-sdk-for-cpp/blob/master/CONTRIBUTING.md#cmake-build-options) to learn about all the supported options and what is the specific option required for each HTTP transport adapter.
+
+Multiple HTTP transport adapters can be built as part of the project. This is to support scenarios where you want to send HTTP request with the libcurl transport adapter for some cases and the WinHTTP transport adapter for others.
+
+Another example is if you want to create your own HTTP transport adapter. In this case, you can build your own HTTP transport adapter and test it by comparing it to the behavior of the libcurl HTTP transport adapter.
+
+## Using the HTTP Transport Adapter
+
+The HTTP transport adapter is set up during service SDK client initialization, for example, when creating an Azure Storage SDK client. The HTTP transport adapter can be specified at client initialization via the client options argument. The next example below shows how to use the default HTTP transport adapter when constructing an SDK client:
+```cpp
+  /*
+  * Using the default HTTP transport adapter.
+  */
+
+  // When no options are provided, the HTTP transport adapter is set to the default one based on the compiler options passed in to CMake.
+  auto defaultStorageClient = BlobServiceClient(url, credential);
+```
+
+The next example below shows how to override the default HTTP transport adapter when constructing an SDK client:
+
+```cpp
+  /*
+  * Override the default HTTP transport adapter.
+  */
+
+  // Use client options to set an HTTP transport adapter, which needs to be a shared_ptr.
+  BlobClientOptions options;
+  options.TransportPolicyOptions.Transport = 
+  // Setting the libcurl transport adapter as an example
+  // Any HTTP transport adapter can be specified here.
+  std::make_shared<Azure::Core::Http::CurlTransport>();
```

Comment:
Move comment above the line rather than in the middle of assignment.

```suggestion
  BlobClientOptions options;
    // Setting the libcurl transport adapter as an example.
  // Any HTTP transport adapter can be specified here.
  options.TransportPolicyOptions.Transport = std::make_shared<Azure::Core::Http::CurlTransport>();
```